R/epi_df.R

```diff
@@ -195,7 +191,7 @@ new_epi_df <- function(x = tibble::tibble(), geo_type, time_type, as_of,
   }
 
   # Apply epi_df class, attach metadata, and return
-  class(x) <- c("epi_df", class(x))
+  class(x) <- c(subclass, "epi_df", class(x))
```

Comment:
question: when is `subclass` useful in a constructor for the main class?  Why wouldn't this just be handled in a separate constructor function?  I see this pattern in dplyr/tibble but also massive extensibility issues in dplyr.  It sort of makes sense in `vctrs::new_vctr` but somehow `new_{tibble,epi_df,epi_archive}` seem different.

musing: especially for `epi_archive`s, except for maybe some role-attachment subclass, I think the extensibility mechanism we most often would want is an "abstract class" that derives certain default operations from certain core operations, but doesn't provide the core operations, + various concrete classes (`{ram,parquet,merge,derived}_epi_archive`) implementing the core operations, or the type-constructor/wrapper thing we do in `grouped_epi_archive`.

Reply:
I'm going to remove these as "not really necessary".